core/app/models/spree/app_configuration.rb

```diff
@@ -309,6 +309,16 @@ def shipping_rate_taxer_class
       @shipping_rate_taxer_class ||= Spree::Tax::ShippingRateTaxer
     end
 
+    # Allows providing your own class for calculating taxes on a shipping rate.
+    #
+    # @!attribute [rw] shipping_rate_tax_calculator_class
+    # @return [Class] a class with the same public interfaces as
+    #   Spree::TaxCalculator::ShippingRate
+    attr_writer :shipping_rate_tax_calculator_class
```

Comment:
Definitely. I've marked both classes and their configuration points with `@api experimental`. (Which is what we used for the `tax_adjuster_class` configuration point.) I also added in a note to each class explaining that they're still in development and that the input specifically is likely to change.